modules/core/src/main/java/org/apache/synapse/mediators/elementary/EnrichMediator.java

```diff
@@ -142,8 +142,19 @@ public boolean mediate(MessageContext synCtx) {
                     isSourcePropertyXML = true;
                 }
             } catch (JsonSyntaxException e) {
-                synLog.traceOrDebug("Source string is not a valid json");
-                isSourcePropertyXML = true;
+                try {
+                    // Enclosing using quotes due to the following issue
+                    // https://github.com/google/gson/issues/1286
+                    String enclosedSourceProperty = "\"" + sourceProperty + "\"";
```

Comment:
Is there any harm in doing this inside the try itself, rather than enclosing with " inside the catch after it fails once?

Reply:
Then there can be a possibility to enclose json objects, json arrays..etc.